src/Credentials/Signer.php

```diff
@@ -33,87 +28,27 @@ class Signer
 	 * @return  Signer  The OAuth message signer object for the message.
 	 *
 	 * @since   1.0
-	 * @throws  InvalidArgumentException
+	 * @throws  \InvalidArgumentException
 	 */
 	public static function getInstance($method)
 	{
 		switch ($method)
 		{
 			case 'HMAC-SHA1':
-				$signer = new SignerHMAC;
+				$signer = new Hmac;
 				break;
 			case 'RSA-SHA1':
 				// @TODO We don't support RSA because we don't yet have a way to inject the private key.
-				throw new InvalidArgumentException('RSA signatures are not supported');
+				throw new \InvalidArgumentException('RSA signatures are not supported');
 				break;
 			case 'PLAINTEXT':
 				$signer = new Plaintext;
 				break;
 			default:
-				throw new InvalidArgumentException('No valid signature method was found.');
+				throw new \InvalidArgumentException('No valid signature method was found.');
 				break;
 		}
 
 		return $signer;
 	}
-
-	/**
-	 * Perform a password authentication challenge.
-	 *
-	 * @param   ClientsTable $client  The client object
-	 * @param   object       $request The Request object.
-	 *
-	 * @return  boolean  True if authentication is ok, false if not
-	 *
-	 * @since   1.0
-	 * @throws  \Exception
-	 */
-	public function doJoomlaAuthentication(ClientsTable $client, $request)
```

Comment:
moved to credentials file - it doesn't have anything to do with signing and was just being proxied anyhow